src/header/common/strict_transport_security.rs

```diff
@@ -39,43 +40,108 @@ use header::{Header, Raw, parsing};
-    pub include_subdomains: bool,
+    include_subdomains: bool,
 
     /// Specifies the number of seconds, after the reception of the STS header
     /// field, during which the UA regards the host (from whom the message was
     /// received) as a Known HSTS Host.
-    pub max_age: u64
+    max_age: u64,
+
+    /// Indicates that the server wants to be included into the HSTS preload list,
+    /// this value can be set to true only if the site:
+    /// 
+    /// * Serves a valid certificate.
+    /// * Redirects all of the traffic on port 80 to port 443.
+    /// * Serves all of its subdomains via HTTPS.
+    /// * Sets in the HSTS headers the parameter includeSubDomains and has max-age
+    ///   bigger than 31536000 seconds (one year).
+    /// * Any redirect will still be served with a HSTS header
+    preload: bool
 }
 
+/// This value indicates the minimum value of the max_age
+/// that can be used togheter with preload
+const PRELOAD_MIN_AGE: u64 = 31536000;
+
 impl StrictTransportSecurity {
-    /// Create an STS header that includes subdomains
-    pub fn including_subdomains(max_age: u64) -> StrictTransportSecurity {
+    /// Create an STS header with a expiration date
+    /// of 5 minutes with no subdomains or preload
+    pub fn new() -> StrictTransportSecurity {
         StrictTransportSecurity {
-            max_age: max_age,
-            include_subdomains: true
+            max_age: 300,
+            include_subdomains: false,
+            preload: false
         }
     }
 
-    /// Create an STS header that excludes subdomains
-    pub fn excluding_subdomains(max_age: u64) -> StrictTransportSecurity {
+    /// Returns the inserted max age
+    pub fn max_age(&self) -> u64 { self.max_age }
+
+    /// Sets the max age
+    pub fn set_max_age(&self, max_age: u64) -> StrictTransportSecurity {
         StrictTransportSecurity {
             max_age: max_age,
-            include_subdomains: false
+            include_subdomains: self.include_subdomains,
+            preload: self.preload
         }
     }
+
+    /// Returns if the header applies HSTS also to subdomains
+    pub fn include_subdomains(&self) -> bool { self.include_subdomains }
+
+    /// Sets if the HSTS header applies also to subdomains
+    pub fn set_include_subdomains(&self, include_subdomains: bool) -> StrictTransportSecurity {
+        StrictTransportSecurity {
+            max_age: self.max_age,
+            include_subdomains: include_subdomains,
+            preload: self.preload
+        }
+    }
+
+    /// Returns if the header applies HSTS preload
+    pub fn preload(&self) -> bool { self.preload }
+
+    /// Sets if the HSTS header applies HSTS preload.
+    /// This will check if the HSTS header satisfies that:
+    /// 
+    /// * It also applies to subdomains
+    /// * It has a max_age greater than 31536000 seconds
+    pub fn set_preload(&self, preload: bool) -> Result<StrictTransportSecurity,()> {
```

Comment:
Hm, I see why this returns a `Result`. Google requires the max age minimum in order for a site to eligible to for its own preload list.

Though, I'm not sure if we should be enforcing that specific requirement here.